node-notes-cli/app.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const dataJson = require('./data.json');
+
+const operation = process.argv[2];
+const choice = process.argv[3];
+
+const jsonNotes = dataJson.notes;
+const notesIndex = Object.keys(jsonNotes);
+
+if (operation === 'read') {
+  fs.readFile('data.json', 'utf8', (err, data) => {
+
+    if (err) throw err;
+
+    for (let i = 0; i < notesIndex.length; i++) {
+      console.log(notesIndex[i] + ': ' + jsonNotes[notesIndex[i]]);
+    }
+  });
+} else if (operation === 'delete') {
+
+  if (notesIndex.includes(choice)) {
+
+    delete jsonNotes[choice];
+
+    const jsonString = JSON.stringify(dataJson, null, 2);
+    fs.writeFile('data.json', jsonString, err => {
+      if (err) throw err;
+    });
+  } else {
+    console.log('Invalid choice');
+  }
+} else if (operation === 'update') {
+
+  if (notesIndex.includes(choice)) {
+
+    const processArray = process.argv;
+    processArray.splice(0, 4);
+
+    const updatedNote = reCreateNote(processArray);
+    jsonNotes[choice] = updatedNote;
+
+    const jsonString = JSON.stringify(dataJson, null, 2);
+    fs.writeFile('data.json', jsonString, err => {
+      if (err) throw err;
+    });
+  } else {
+    console.log('Invalid choice');
+  }
+} else if (operation === 'create') {
+  const processArray = process.argv;
+  processArray.splice(0, 3);
```

Comment:
It's generally better to create a new array with `.slice()` than to modify the original array with `.splice()`.
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/slice